ext/reflection/php_reflection.c

```diff
@@ -3777,8 +3778,36 @@ ZEND_METHOD(reflection_class, getStaticProperties)
 		return;
 	}
 
+	if (!CE_STATIC_MEMBERS(ce)) {
+		zend_class_init_statics(ce);
+	}
+
 	array_init(return_value);
-	add_class_vars(ce, 1, return_value);
+
+	ZEND_HASH_FOREACH_STR_KEY_PTR(&ce->properties_info, key, prop_info) {
+		zval *members;
+
+		if (((prop_info->flags & ZEND_ACC_PRIVATE) &&
+		     prop_info->ce != ce)) {
+			continue;
+		}
+		if ((prop_info->flags & ZEND_ACC_STATIC) == 0) {
+			continue;
+		}
+
+		prop = &CE_STATIC_MEMBERS(ce)[prop_info->offset];
+		ZVAL_DEINDIRECT(prop);
+
+		if (prop_info->type && Z_ISUNDEF_P(prop)) {
+			continue;
+		}
+
+		/* copy: enforce read only access */
+		ZVAL_DEREF(prop);
+		ZVAL_COPY_OR_DUP(&prop_copy, prop);
```

Comment:
The CE_STATIC_MEMBERS is already per-request, so it's not necessary to COPY_OR_DUP. Just doing a `Z_TRY_ADDREF_P(prop)` is sufficient here.

Reply:
Fine! :)